tests/integration/replication.tcl

```diff
@@ -994,3 +994,37 @@ start_server {tags {"repl external:skip"}} {
         }
     }
 }
+
+test {replica can handle EINTR if use diskless load} {
+    start_server {tags {"repl"}} {
+        set slave [srv 0 client]
+        set slave_log [srv 0 stdout]
+        start_server {} {
+            set master [srv 0 client]
+            set master_host [srv 0 host]
+            set master_port [srv 0 port]
+
+            $master debug populate 5 master 100000
+            $master config set rdbcompression no
+            $master config set repl-diskless-sync yes
+            $master config set repl-diskless-sync-delay 0
+            $slave config set repl-diskless-load on-empty-db
+            # Construct EINTR error
+            $slave config set watchdog-period 200
+            # Block slave in read()
+            $master config set rdb-key-save-delay 1000000
+            # Start the replication process...
+            $slave slaveof $master_host $master_port
+
+            # wait for the slave to start reading the rdb
+            wait_for_condition 50 100 {
+                [s -1 master_link_status] eq {up}
```

Comment:
i now notice this this won't wait for the salve to start reading, it'll wait for it to finish reading (so this test will be slow).
instead i would like to propose to wait for `[s -1 loading] eq 1`, or better yet, look at the log file so we can make sure the `WATCHDOG TIMER EXPIRED` message is seen after `Loading DB in memory`.

have a quick look at `tests/integration/replication.tcl` you can see an example of how this is done using `wait_for_log_messages` there.

Reply:
@menwenjun can you please try to take the second approach i suggested above?
i think i'll be better in both making sure this test doesn't fail due to timing issues (if loading is detected to start before the watchdog fired), and also to make sure the test doesn't succeed even when there was a signal received before loading started, but not after (failing to test what we meant for)

Reply:
@oranagra I used `wait_for_log_messages`, please CR. thank you!